src/ofxCcv.cpp

```diff
@@ -122,24 +152,39 @@ vector<ofImage> ofxCcv::getWeights() {
     return weightImgs;
 }
 
-vector<float> ofxCcv::encode(ofPixels & img, int layer) {
+vector<float> ofxCcv::encode(const ofPixels& pix, int layer) const {
     convnet->count = layer; // hack to extract a particular layer with encode
-    vector<float> data;
-    ofImage imgCopy;
-    imgCopy.setFromPixels(img);
-    imgCopy.resize(224, 224); // resize to AlexNet input size (224x224)
+
     ccv_dense_matrix_t image;
-    image = toCcv(imgCopy);
-    ccv_dense_matrix_t* input = 0;
-    ccv_size_t size = ccv_size(225, 225);   // size of the default net, this may need to be changed
+
+    if (pix.getWidth() != 224 || pix.getHeight() != 224)
```

Comment:
Avoid any allocations at all if the pixels are already the correct size.  Resize is prevented inside of `ofImage::resize` but this prevents copies.